modules/programs/exa.nix

```diff
@@ -23,10 +11,49 @@ in {
 
     enableAliases = mkEnableOption "recommended exa aliases";
 
+    extraOptions = mkOption {
+      type = types.listOf types.str;
+      default = [ ];
+      example = [ "--group-directories-first" "--header" ];
+      description = ''
+        Extra command line options passed to exa.
```

Comment:
Maybe worth mentioning that this only applies to aliases defined in the module? Would it make sense to make the options apply in all cases? E.g., by wrapping the original `exa` executable by one that adds the options.

Reply:
Hmm. There's a chance for that to mess with e.g. scripts that use `exa` internally and expect it to work like vanilla `exa`. But then again, I don't think it's commonly used in scripts since there is basically no upside to that compared to just using `ls` in that use-case. I added wrapping -- let's see if anyone complains about it later. :)

Reply:
I think you don't want to hard-code a binary with options. I think usually it's done like this:

```bash
alias ls="exa ${extra_options}"
alias ll="ls -l"
alias la="ls -la"
```

This way user (or script) can always run `\ls`, `/bin/ls`, `command ls`  or whatever to ignore any potential aliases or options.

Reply:
I agree. As a user I expect aliases to be made, not the 'binary' to be changed.

Reply:
There seems to be precedent for both variants within home-manager. I agree that the variant with just adding a shadowing alias is probably preferable. I force-pushed again with an amended version that implements this instead.